Select operation revision

jdanieck opened the pull request:
### Details:
 - Added Serialization SLT
 - Added Visitor tests
 - Added Select-1 to summarize.py
 - Remove Select from evaluates_map
 - Migrated backend tests to template plugin
 - Remove Select tests from manifest
 - Refactored CoordinateTransform usage to "CoordinateTransformBasic/row_major_strides" pattern (benchmarks results in the ticket)
- **PDPD broadcasting implementation aligned to [spec](https://github.com/openvinotoolkit/openvino/pull/6488)** -  new order of tensor broadcasting is *then* <- *else* <- *cond*; before was *else* <- *if* <- *cond*;

### Tickets:
 - *37531*

jdanieck commented:
TODO: remove CoordinateTransform usage from [autobroadcast_select](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/reference/include/ngraph/runtime/reference/autobroadcast_binop.hpp#L354).

jdanieck commented:
TODO: add more unit tests (template_plugin)

pelszkow commented on docs/template_plugin/tests/functional/op_reference/select.cpp:
```diff
@@ -0,0 +1,139 @@
+
+struct SelectParams {
+    template <class IT, class OT>
+    SelectParams(const ngraph::element::Type& data_type, const ngraph::op::AutoBroadcastSpec& broadcast, const ngraph::PartialShape& select_input_pshape,
```
Above is `using namespace ngraph;`. Why types are prefix by `ngraph::`?

jdanieck replied:
It's unnecessary, removed.

pelszkow commented on docs/template_plugin/tests/functional/op_reference/select.cpp:
```diff
@@ -0,0 +1,139 @@
+INSTANTIATE_TEST_SUITE_P(smoke_Select_With_Hardcoded_Refs, ReferenceSelectLayerTest,
+                         ::testing::Values(
+                             // fp32, no brodcasting
+                             SelectParams(element::f32,                                         // if/else/output data type
```
Consider builder instead o comments (and c-tor) e.g.: https://github.com/openvinotoolkit/openvino/pull/6623/files#diff-332d725406461a0f91052683c0660e5eee5616601c83aa7e91a5da03bb5f64bdR150-R156

jdanieck replied:
Clever approach, but I will stick to comments this time :)

jdanieck commented:
@ilyachur could you find someone who is familiar with PDPD broadcasting to review this PR? I changed order of tensor broadcasting to be aligned with spec and with reference implementation. It was not tested before and when I added unit tests I discovered it is not working (in my understanding). This can break backward compatibility, so we need to asses this change carefully.

jdanieck commented:
@pelszkow please take care of this PR starting from Monday as I am out of project starting next week. I'll be available to all consultations you need, just won't be making code changes anymore.
